src/payment/instrument/instrument-action-creator.ts

```diff
@@ -125,10 +125,12 @@ export default class InstrumentActionCreator {
 
         const { customerId } = cart;
         const { storeId } = config.storeProfile;
+        const { code } = config.shopperCurrency;
```

Comment:
The active shopper currency does not necessarily have to match the currency that the stored instrument was created with. Also, `shopperCurrency` could be a display currency.

However, I'm still thinking that if the instrument ID is unique, we shouldn't need the currency code to delete it. If it's a 1-1 relationship, I would encourage to drop the requirement for a currency code when deleting an instrument.

If you still think that we need to pass currency code to delete an instrument, probably the correct way would involve modifying the instruments endpoint so each instrument returns the currency linked with each instrument.